### PR TITLE
CLN: clean libreduction

### PR DESCRIPTION
**Pull request by fangchenli**

### CODE REVIEW

**Comment by jreback** [comment]
sure, no perf change?

File: pandas/_libs/reduction.pyx [hunk 1]
```diff
@@ -408,29 +402,24 @@ cdef class BlockSlider:
...
-        cdef:
-            Py_ssize_t i
-            object b
-
         self.frame = frame
         self.dummy = frame[:0]
         self.index = self.dummy.index
 
-        self.blocks = [b.values for b in self.dummy._mgr.blocks]
+        self.blocks = [block.values for block in self.dummy._mgr.blocks]
```
Comment (jbrockmendel) [inline pandas/_libs/reduction.pyx:419]: can we call this self.blk_values?  calling it blocks is misleading

File: pandas/_libs/reduction.pyx [hunk 2]
```diff
@@ -467,11 +453,7 @@ cdef class BlockSlider:
         self.index._cache.clear()  # e.g. inferred_freq must go
 
     cdef reset(self):
-        cdef:
-            ndarray arr
-            Py_ssize_t i
-
-        # reset blocks
+        cdef ndarray arr
```
Comment (jbrockmendel) [inline pandas/_libs/reduction.pyx:456]: we pretty consistently us the two-line version

```
cdef:
    ndarray arr
```

File: pandas/_libs/reduction.pyx [hunk 3]
```diff
@@ -439,18 +428,15 @@ cdef class BlockSlider:
...
         for i, block in enumerate(self.blocks):
             self.base_ptrs[i] = (<ndarray>block).data
 
     def __dealloc__(self):
         free(self.base_ptrs)
 
     cdef move(self, int start, int end):
-        cdef:
-            ndarray arr
-            Py_ssize_t i
```
Comment (jbrockmendel) [inline pandas/_libs/reduction.pyx:452]: does typing i not help?

**Comment by fangchenli** [comment]
> sure, no perf change?

No perf changes on groupby.

**Comment by jreback** [comment]
thanks @fangchenli

**Pull request merged**